### ISSUE TITLE
Move to Snakemake 8.0

### ISSUE THREAD

**Issue by LeeBergstrand** [jmtsuji/rotary#111]
@jmtsuji So Snakemake 8.0 just came out.

https://snakemake.readthedocs.io/en/stable/getting_started/migration.html

It has a lot of changes and breaking Syntax.

Note that it requires Python 3.11. However, Circulator requires pysam, which may have compatibility issues with newer versions of Python. 

I think that once we move away from Circulator, we can move to Snakemake 8.0.

**Comment by jmtsuji** [comment]
Agreed -- let's plan to move to Snakelike 8.0 after moving away from Circulator

**Comment by LeeBergstrand** [comment]
@jmtsuji I had to downgrade back to Snakemake 7 due to an issue I found in Snakemake 8 see https://github.com/rotary-genomics/rotary/pull/217

See https://github.com/rotary-genomics/rotary/pull/214 for reasoning.
